kolibri/core/auth/models.py

```diff
@@ -160,7 +177,12 @@ class FacilityDataset(FacilityDataSyncableModel):
     learner_can_delete_account = models.BooleanField(default=True)
     learner_can_login_with_no_password = models.BooleanField(default=False)
     show_download_button_in_learn = models.BooleanField(default=True)
-    extra_fields = JSONField(null=True, blank=True)
+    extra_fields = JSONField(
+        null=True,
+        blank=True,
+        validators=[JSON_Schema_Validator(extra_fields_schema)],
```

Comment:
Nice! I like the use of a schema validator here :100:

Reply:
If the model is saved with `extra_fields` missing the new `pin_code` field, does it pass validation?

Reply:
Validation will fail with a `400`. Generally one of fields should be present for validation to pass, preventing unknown fields from being saved. There should be a test covering this scenario

Reply:
So if an existing database is upgraded to 0.16, and they haven't set a pin so the field doesn't exist, will other saves to the model fail because of that?

Reply:
I don't think so as the `extra_fields` fields is an optional field and is `null` by default. But also, default values have been defined for the fields just in case.